src/main/client/player/client-player-manager.ts

```diff
@@ -18,5 +18,9 @@ export class ClientPlayerManager {
             position: { x: 0, y: 0 },
          });
       });
+
+      component.clientShooting$.subscribe((shootingOptions) => {
+         network.sendShootRequest(shootingOptions);
+      });
```

Comment:
The manager should be the other way around. By following the manager-component structure (might not the best and could be improved in the future, but at the moment) the ClientNetworkManager should subscribe to the ClientPlayerComponent.clientShooting$ event to call component.sendShootRequest.

Reply:
`sendShootRequest` is on `network` though
should I move this to method to `component.sendShootRequest`? dont fully understand

Reply:
Move this scubscribtion to ClientNetworkManager and use ClientPlayerComponent.clientShooting$.subscribe((options)=>component.sendShootRequest(options))

Reply:
ok done

but then what is the purpose of the ClientPlayerManager? if it does not manage the player?

Reply:
It can. It can subscribe to other components' events. There is already one subscription.